File: 001.c
#include <stdio.h>

int
main (void) {
    int total = 0;

    for (int i = 0; i < 1000; i++) {
        if (i % 3 == 0 || i % 5 == 0)
            total += i;
    }
    printf("%d\n", total);
    return 0;
}

File: 001.scala
// Imperative Style
def threeOrFive(x: Int) = if ((x % 3 == 0) || (x % 5 == 0)) true else false
var total = 0

for (i <- 3 to 999) {
  if (threeOrFive(i))
    total += i
}

println(total)

// Functional style
total = (3 to 999).filter(s => (s % 3 == 0 || s % 5 == 0)).sum
println(total)
